File: Char_Detector/char_detector.cpp
#include <iostream>
#include <string>
using namespace std;

int charDetector(string sentence, char searchLetter)
{
  int sentenceLength = sentence.length();
  int letterInstances = 0;

  for(int i = 0; i < sentenceLength; i++){
    if(char(sentence[i]) == searchLetter) letterInstances++;
  }

  return letterInstances;
}

int main()
{
  string sentence = "I am the very model of a modern major general.";
  char searchLetter = 'e';

  cout << "Starting sentence: " << sentence << endl;
  int letterInstances = charDetector(sentence, searchLetter);
  cout << searchLetter << " is featured " << letterInstances << " times." << endl;

  return 0;
}

File: Char_Detector/CharDetector.java
// This is the java implementation of the CharDetector program.

import java.lang.*;

public class CharDetector {
  static int charDetector(String sentence, char searchLetter) {
    int letterInstances = 0;
    for(int i = 0; i < sentence.length(); i++) {
      if(Character.toLowerCase(sentence.charAt(i)) == Character.toLowerCase(searchLetter)) letterInstances++;
    }
    return letterInstances;
  }

  public static void main(String[] args) {
    String sentence = "I am the very model of a modern major general.";
    char searchLetter = 'e';

    System.out.println("Starting sentence: " + sentence);
    int letterInstances = charDetector(sentence, searchLetter);
    System.out.println(searchLetter + " is featured " + letterInstances + " times.");
  }
}
